LoginClient/PipeClient.cs

```diff
@@ -0,0 +1,368 @@
+        {
+            return pipeName;
+        }
+
+        /// <summary>
+        /// Set the name of the pipe that connects to the server.
+        /// </summary>
+        /// <param name="newPipeName">The new pipe name.</param>
+        private void SetPipeName(string newPipeName)
+        {
+            pipeName = newPipeName;
+        }
+
+        #endregion
+
+
+        #region Connection
+
+        /// <summary>
+        /// Disconnects the client from the server.
+        /// </summary>
+        public void Disconnect()
+        {
+            if(!IsConnected())
+            {
+                return;
+            }
+
+            // We're no longer connected to the server.
+            IsConnected(false);
+            SetPipeName(null);
+            IsLoggedIn(false);
+
+            readThread.Abort();
+
+            // Clean up the resources
+            if (stream != null)
+            {
+                stream.Close();
+            }
+            handle.Close();
+
+            stream = null;
+            handle = null;
+        }
+
+        /// <summary>
+        /// Connects client to the server with the pipename.
+        /// </summary>
+        /// <param name="pipename">The name of the pipe to connect to.</param>
+        public void Connect(string pipename)
+        {
+            if (IsConnected())
+            {
+                throw new Exception("Already connected to server.");
+            }
+
+            SetPipeName(pipename);
+
+            handle = CreateFile(
+                GetPipeName(),
+                0xC0000000, // GENERIC_READ | GENERIC_WRITE = 0x80000000 | 0x40000000
+                0,
+                IntPtr.Zero,
+                3,  // OPEN_EXISTING
+                0x40000000, // FILE_FLAG_OVERLAPPED
+                IntPtr.Zero);
+
+            // Couldn't create a handle.
+            // The server is probably not running.
+            if (handle.IsInvalid)
+            {
+                MessageBox.Show(
+                    "Unable to connect.\nTheServer may not be running.",
+                    "Server connection Error.");
+                return;
+            }
+
+            stream = new FileStream(handle, FileAccess.ReadWrite, BUFFER_SIZE, true);
+
+            IsConnected(true);
+
+            // Start listening for messages.
+            readThread = new Thread(Read)
+            {
+                IsBackground = true
+            };
+            readThread.Start();
+        }
+
+        #endregion
+        /// <summary>
+        /// Takes the byte[] from the server and checks the login attempt was successful.
+        /// </summary>
+        /// <param name="result">Byte[] </param>
+        private void ValidationResult(byte[] result)
+        {
+            ASCIIEncoding encoder = new ASCIIEncoding();
+            string str = encoder.GetString(result, 0, result.Length);
+
+            if (str.Equals("You are now logged in."))
+            {
+                IsLoggedIn(true);
```

Comment:
Bug.
Your already connected to the server, but IsLoggedIn, connects to the server.
See IsLoggedIn Comment for recommended fix to make this clear.